### PR TITLE
 adding field and block classes

### PR DESCRIPTION
**Pull request by SNsir**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 1]
```diff
@@ -284,4 +292,4 @@ abstract class VField(val width: Int, val height: Int) {
   private val background // background colors for this field.
           : Array<VColor>
 
-}
+}
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:295]: add empty line

File: galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt [hunk 2]
```diff
@@ -15,7 +15,127 @@
...
+   */
+  fun getModel(): VField = rowController.getModel()
+
+  //-----------------------------------------------------------------------
+  // FIELDHANDLER IMPLEMENTATION
+  // ----------------------------------------------------------------------
+
+  override fun getRowController(): VFieldUI {
+    return rowController
+  }
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt:45]: ` override fun getRowController(): VFieldUI = rowController`

**Review by h-haddad — APPROVED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/form/Alignment.kt [hunk 3]
```diff
@@ -30,7 +30,7 @@ import java.io.Serializable
  * @param          alignRight      position in alignRight
  * @param          useAll          use the whole possible width of the column
  */
-class Alignment(var x: Int,
+open class Alignment(var x: Int,
                 var y: Int,
                 var width: Int,
                 var height: Int,
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/Alignment.kt]: check indentation

File: galite-core/src/main/kotlin/org/kopi/galite/form/VForm.kt [hunk 4]
```diff
@@ -18,9 +18,18 @@
 
 package org.kopi.galite.form
 
+import org.kopi.galite.visual.VActor
 import org.kopi.galite.visual.VWindow
 
-abstract class VForm : VConstants, VWindow() {
+abstract class VForm : VConstants,VWindow(){
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VForm.kt:24]: add empty space between VConstantsand VWindow()

File: galite-core/src/main/kotlin/org/kopi/galite/visual/Module.kt [hunk 5]
```diff
@@ -18,8 +18,44 @@
 
 package org.kopi.galite.visual
 
-open class Module {
+import org.kopi.galite.l10n.LocalizationManager
+
+class Module(
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/Module.kt]: check indentation

**Review by h-haddad — APPROVED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt [hunk 6]
```diff
@@ -15,7 +15,125 @@
...
+abstract class AbstractFieldHandler protected constructor(private val rowController: VFieldUI)
+  : FieldHandler {
+
+  // ----------------------------------------------------------------------
+  // MODEL ACCESSOR
+  // ----------------------------------------------------------------------
+
+  /**
+   * Returns the row controller field getModel().
+   * @return The row controller field getModel().
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt:35]: `field model`

File: galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt [hunk 7]
```diff
@@ -15,7 +15,125 @@
...
+
+  override fun fieldError(message: String) {
+    rowController.displayFieldError(message)
+
+    fun requestFocus(): Boolean {
+      rowController.transferFocus(getCurrentDisplay()!!)
+      return true
+    }
+
+    fun loadItem(mode: Int): Boolean {
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt:87]: I dont agree

File: galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt [hunk 8]
```diff
@@ -15,7 +15,125 @@
...
+      var id = -1
+      val dictionary = (if (getModel().list != null && getModel().list!!.newForm != null) {
+        // OLD SYNTAX
+        Module.getExecutable(getModel().list!!.newForm) as VDictionary
+      } else if (getModel().list != null && getModel().list!!.action != -1) {
+        // NEW SYNTAX
+        getModel().block.executeObjectTrigger(getModel().list!!.action) as VDictionary
+      } else {
+        null
+      })
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt:99]: change it to `when` statement

File: galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt [hunk 9]
```diff
@@ -15,7 +15,125 @@
...
+  override fun getCurrentDisplay(): UField? = rowController.getDisplay()
+
+  override fun fieldError(message: String) {
+    rowController.displayFieldError(message)
+  }
+
+    override fun requestFocus(): Boolean {
+      rowController.transferFocus(getCurrentDisplay()!!)
+      return true
+    }
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt:86]: indentation

File: galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt [hunk 10]
```diff
@@ -15,7 +15,125 @@
...
+
+    override fun loadItem(mode: Int): Boolean {
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt:88]: indentation

File: galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt [hunk 11]
```diff
@@ -15,7 +15,125 @@
...
+      val dictionary = (if (getModel().list != null && getModel().list!!.newForm != null) {
+        // OLD SYNTAX
+        Module.getExecutable(getModel().list!!.newForm) as VDictionary
+      } else if (getModel().list != null && getModel().list!!.action != -1) {
+        // NEW SYNTAX
+        getModel().block.executeObjectTrigger(getModel().list!!.action) as VDictionary
+      } else {
+        null
+      })
+              ?: return false
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/AbstractFieldHandler.kt:100]: move this at the end of previous line:
`}) ?: return false`

File: galite-core/src/main/kotlin/org/kopi/galite/form/AbstractPredefinedValueHandler.kt [hunk 12]
```diff
@@ -15,7 +15,29 @@
...
  */
+
 package org.kopi.galite.form
 
-abstract class AbstractPredefinedValueHandler {
+import org.kopi.galite.list.VListColumn
+
+abstract class AbstractPredefinedValueHandler(private val model: VFieldUI,
+                                              protected var form: VForm,
+                                              protected var field: VField)
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/AbstractPredefinedValueHandler.kt:25]: `val`

File: galite-core/src/main/kotlin/org/kopi/galite/form/BlockAlignment.kt [hunk 13]
```diff
@@ -15,7 +15,40 @@
...
 package org.kopi.galite.form
 
-class BlockAlignment {
+import java.io.Serializable
+
+/**
+ * A class to specify alignment in Layout
+ * Creates a new `BlockAlignment` instance.
+ *
+ * @param          block          Represents the blockginal block to be aligned with.
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/BlockAlignment.kt:27]: fix this doc comment
`the original block`

**Pull request merged**